src/main/java/org/broadinstitute/hellbender/utils/iterators/AlignmentContextIteratorFactory.java

```diff
@@ -0,0 +1,113 @@
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.engine.AlignmentContext;
+import org.broadinstitute.hellbender.engine.ReferenceDataSource;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.locusiterator.LIBSDownsamplingInfo;
+import org.broadinstitute.hellbender.utils.locusiterator.LocusIteratorByState;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.stream.Collectors;
+
+/**
+ * Create an iterator for traversing alignment contexts in a specified manner.  This class should be able to support both spark and non-spark LocusWalker
+ */
+public class AlignmentContextIteratorFactory {
+
+    protected static final Logger logger = LogManager.getLogger(AlignmentContextIteratorFactory.class);
+
+    private AlignmentContextIteratorFactory() {}
+
+    /**
+     *  Create the appropriate instance of an alignment context spliterator based on the input parameters.
+     *
+     *  Please note that this wrapper is still tied to {@link LocusIteratorByState} and some parameters are being passed directly to that class.
+     *
+     * @param intervalsForTraversal the intervals to generate alignment contexts over.
+     * @param header SAM file header to use
+     * @param readIterator iterator of sorted GATK reads
+     * @param dictionary the SAMSequenceDictionary being used for this traversal.  This can be the same as the reference.  {@code null} is supported, but will often lead to invalid parameter combinations.
+     * @param downsamplingInfo how to downsample (for {@link LocusIteratorByState})
+     * @param reference the reference being used for this traversal, {@code null} if no reference being used.
+     * @param emitEmptyLoci whether loci with no coverage should be emitted.  In this case, the AlignmentContext will be empty (not null).
+     * @param isKeepUniqueReadListInLibs if true, we will keep the unique reads from the samIterator and make them
+     *                                       available via the transferReadsFromAllPreviousPileups interface (this parameter is specific to {@link LocusIteratorByState})
+     * @param isIncludeDeletions include reads with deletion on the loci in question
+     * @param isIncludeNs include reads with N on the loci in question
+     * @return Spliterator that produces AlignmentContexts ready for consumption (e.g. by a {@link org.broadinstitute.hellbender.engine.LocusWalker})
+     */
+    public static Iterator<AlignmentContext> createAlignmentContextIterator(final List<SimpleInterval> intervalsForTraversal,
```

Comment:
This gives me default values for the creation, but I can get that with overloading.  If I am keeping a state that can be changed, it would be a ``Builder``, right?

I'm open to the idea, but need a little more convincing.